packages/gatsby/src/bootstrap/load-themes/index.js

```diff
@@ -5,6 +5,34 @@ const _ = require(`lodash`)
 const debug = require(`debug`)(`gatsby:load-themes`)
 const preferDefault = require(`../prefer-default`)
 const getConfigFile = require(`../get-config-file`)
+const {
+  validatePluginOptions,
+  formatOptionsError,
+} = require(`../../utils/validate-plugin-options`)
+const reporter = require(`gatsby-cli/lib/reporter`)
+const Joi = require(`@hapi/joi`)
+
+const getPluginOptions = theme => {
```

Comment:
I kinda hate this -- the idea is that we need to replicate the coercion that the validatePluginOptions API will normally do.

However -- themes are a special kind of plugin (that can create other plugins) so we have to do some hackery here to ensure that options are coerced (and thrown) and passed to the theme as expected.

Suggestions welcome!